test/system/src/test/java/io/pravega/test/system/ReadWriteAndAutoScaleWithFailoverTest.java

```diff
@@ -94,6 +94,8 @@ public void setup() {
         //get Controller Uri
         controller = new ControllerImpl(controllerURIDirect);
         testState = new TestState();
+        testState.writersListComplete.set(0, testState.writersComplete);
```

Comment:
This is going to throw as we need to `add` first before invoking `set`. `set` is used to replace an element, but there must have been some add in that position before. Let's replace this with a call to `add`. I don't think there is a need to invoke `set` here.